src/npmUtils.ts

```diff
@@ -0,0 +1,50 @@
+import fs from "fs";
+import * as ini from "ini";
+
+export const prepareNpmConfig = (
+  // Allow to inject a custom object, useful in tests.
+  processEnv = process.env
+) => {
+  const userNpmrcPath = `${processEnv.HOME}/.npmrc`;
+
+  if (fs.existsSync(userNpmrcPath)) {
+    console.log(`Found existing user .npmrc file at ${userNpmrcPath}`);
+
+    // Parse the `.npmrc` content using the `npm/ini` package.
+    const npmConfig = ini.parse(fs.readFileSync(userNpmrcPath, "utf-8"));
+
+    let hasAuthToken = false;
+    for (const [key, value] of Object.entries(npmConfig)) {
+      if (/\/\/(.*)authToken$/.test(key) && Boolean(value)) {
+        console.log("The .npmrc file has an authToken");
+        hasAuthToken = true;
+      }
+    }
```

Comment:
This is the "new logic" to test the presence of the `authToken`. The regex also can be simpler.